tests/conftest.py

```diff
@@ -58,6 +58,432 @@ def pytest_logger_logdirlink():
     return log_path
 
 
+@pytest.fixture()
+def rbd_secret_factory(request):
```

Comment:
I asked the same question on #428 but is `{resource}_factory` the naming convention we want to go with for factory fixtures? I think they would be better as `create_{resource}` so when a test or other fixture utilizes this fixture it would look like this:

```python
def my_test(create_rbd_secret):
    my_secret = create_rbd_secret()
```
instead of this:
```python
def my_test(rbd_secret_factory):
    my_secret = rbd_secret_factory()
```

It's not a huge deal but I think `create_{resource}` reflects more accurately what the fixture is actually doing.

Reply:
I agree that it would look better but this convention is currently used in `helper.py` and maybe on some other place. I would stick with `{resource}_factory` convention as it doesn't conflict with any other functions. If we would want to use `create_{resource}` we would need to ensure that this convention is used only for factory methods before we use it.

Reply:
Yeah I can see that being an issue, that's a good point. I'm fine either way, let's just make sure we are consistent in #428. (sorry @petr-balogh for the back and forth on this)

Reply:
OK, so do we agree on {resource}_factory? I have another idea about create_{resource}_factory as the name of the factory fixture will be very similar to helper function used in it most cases so we know that it's probably returning the fucntion create_{resource} and we have the factory so we know it's factory fixture.  WDYT?

Reply:
`create_{resource}_factory` seems little too long to me and I don't think it provides any extra information (the factory can be created even without create helper function). I would leave it as `{resource}_factory` for now and in future we can try to get rid of helpers and turn them into factories so we can rename all factories to `create_{resource}`.